autospec/pkg_integrity.py

```diff
@@ -210,32 +206,43 @@ def __del__(self):
         print(SEPT)
 
 
-def head_request(url):
-    curl = pycurl.Curl()
-    curl.setopt(curl.URL, url)
-    curl.setopt(curl.CUSTOMREQUEST, "HEAD")
-    curl.setopt(curl.NOBODY, True)
-    curl.setopt(curl.TIMEOUT, 5)
-    curl.setopt(curl.FOLLOWLOCATION, True)
-    curl.perform()
-    http_code = curl.getinfo(pycurl.HTTP_CODE)
-    curl.close()
-    return http_code
+def download_file(url, destination):
+    sign_file = os.path.join(destination, os.path.basename(url))
+    print("Attempting {}".format(sign_file))
+    with open(sign_file, 'wb') as f:
+        curl = pycurl.Curl()
+        curl.setopt(curl.URL, url)
+        curl.setopt(curl.WRITEDATA, f)
+        curl.setopt(curl.FOLLOWLOCATION, True)
+        try:
+            curl.perform()
+        except pycurl.error as e:
+            print(e.args)
+            return None
+        code = curl.getinfo(pycurl.HTTP_CODE)
+        curl.close()
+        if code != 200:
+            os.unlink(sign_file)
+            return None
+        print("Downloaded {}!".format(sign_file))
+        return sign_file
+    return None
 
 
-def get_signature_url(package_url):
+def get_signature_file(package_url, package_path):
     if 'samba.org' in package_url:
-        return package_url + '.asc'
+        return download_file(package_url + '.asc', package_path)
     elif '://pypi.' in package_url[:13]:
-        return package_url + '.asc'
+        return download_file(package_url + '.asc', package_path)
     elif 'mirrors.kernel.org' in package_url:
-        return package_url + '.sig'
+        return download_file(package_url + '.sig', package_path)
```

Comment:
I'm ambivalent about keeping these and just letting any type be tried. I don't see it adding a major slowdown especially as it is only downloaded once.

Reply:
I'll keep this logic under pretense that it might be restricting the types of signatures that are expected that specific location.